Doc/glossary.rst

```diff
@@ -231,6 +231,26 @@ Glossary
       A variable defined in a class and intended to be modified only at
       class level (i.e., not in an instance of the class).
 
+   closure variable
+      A :term:`free variable` referenced from a :term:`nested scope` that is defined in an outer
+      scope rather than being resolved at runtime from the globals or builtin namespaces.
+      May be explicitly defined with the :keyword:`nonlocal` keyword to allow write access,
+      or implicitly defined if the variable is only read without being written to.
+
+      For example, in the ``inner`` function in the following code, both ``x`` and ``print`` are
+      :term:`free variables <free variable>`, but only ``x`` is a *closure variable*::
+
+          def outer():
+              x = 0
+              def inner():
+                  nonlocal x
+                  x += 1
+                  print(x)
+              return inner
+
+      Due to the name of the :attr:`codeobject.co_freevars` attribute, sometimes instead
+      referred to as a :term:`free variable`.
```

Comment:
```suggestion
      Closure variables are sometimes
      referred to as a :term:`free variable` due to the name of the :attr:`codeobject.co_freevars` attribute.
```

Reply:
I realised this wasn't quite saying what I meant. Calling a closure variable a free variable is fine, since closure variables are a subset of free variables. The misleading phrasing is when people say "free variables" when only referring to a function's "closure variables".

```suggestion
      Due to the :attr:`codeobject.co_freevars` attribute (which, despite its name, only
      includes the names of closure variables rather than listing all referenced free variables),
      the more general :term:`free variable` term is sometimes used even when the intended
      meaning is to refer specifically to closure variables.
```

Reply:
I think this phrasing is still awkward, but it's at least explicit as to what I meant. If anyone comes up with some better wording later, we'll at least have a framework to include it now.

Reply:
Love the explicit wording. I think that works well @ncoghlan.